dotnet/src/Agents/Abstractions/AgentChannel.cs

```diff
@@ -2,15 +2,22 @@
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
 
 namespace Microsoft.SemanticKernel.Agents;
 
 /// <summary>
 /// Defines the communication protocol for a particular <see cref="Agent"/> type.
-/// An agent provides it own <see cref="AgentChannel"/> via <see cref="Agent.CreateChannelAsync(CancellationToken)"/>.
+/// An agent provides it own <see cref="AgentChannel"/> via <see cref="Agent.CreateChannelAsync"/>.
 /// </summary>
 public abstract class AgentChannel
 {
+    /// <summary>
+    /// The <see cref="ILogger"/> associated with the <see cref="AgentChannel"/>.
+    /// </summary>
+    public ILogger Logger { get; set; } = NullLogger.Instance;
```

Comment:
I like that it's on class level here, but wondering why not in `Agent.cs`.

Reply:
Channel is where all the dynamism is.  (Similiar to strategy.)  Agent doesn't have much interesting to say other than maybe when its creating a channel.